test/unittest/layers/layers_common_tests.h

```diff
@@ -79,11 +79,24 @@ class LayerSemantics
   bool must_fail;
 };
 
+typedef enum {
+  SKIP_CALC_GRAD = 1 << 0,  /**< skip calculating gradient and compare */
+  SKIP_CALC_DERIV = 1 << 1, /**< skip calculating derivative and compare */
+
+  FORWARD_MODE_INFERENCE =
+    1 << 2, /**< set if layer should be forwarded with inference mode */
+
+  DEFAULT = 0, /**< default set up */
```

Comment:
is default only set up or setup and run both forwarding and backwarding?

Reply:
Done :)